### PR TITLE
V2:Astro-961 - Tree KM Style Update

### PR DESCRIPTION
**Pull request by im-cr**
**Tickets**
  - https://rocketcom.atlassian.net/browse/ASTRO-961

**Changes** - commit: 
  - rux-tree.js
  - theme--dark
  - theme--light
  - tree.css

**Designs**
  - [Dark](https://app.abstract.com/projects/2d8cf270-0b93-11e8-89ff-61554354acb1/branches/d80b7c6b-5773-4fc5-9040-5d4e98eb83e8/files/1d17329d-e1a3-490a-a6ad-097e2ecd4825?filter=changed)
  - [Light](https://app.abstract.com/projects/2d8cf270-0b93-11e8-89ff-61554354acb1/branches/d80b7c6b-5773-4fc5-9040-5d4e98eb83e8/files/b7c2bf0c-9dec-4b07-96d4-ead85393c9d0?filter=changed)

### CODE REVIEW

**Comment by amalyah** [comment]
Design has a primary border around the outside of the full component (not just top and bottom borders).

**Comment by amalyah** [comment]
Do we allow for tertiary states of groupings in the code? (We do in design, so I was curious about the code.)

**Comment by amalyah** [comment]
Hover state should be a background color adjustment like other components. (primary-dark for light theme and primary-light for dark theme.)

File: src/css/common/_theme--dark.css [hunk 1]
```diff
@@ -269,23 +269,23 @@
...
+  --treeBorderColor: var(--primaryDark);
+  --treeItemBorderColor: var(--primaryDark);
   --treeAccentColor: var(--primary);
 
   /* --treeFocusedBackgroundColor: var(--colorTertiaryLighten1, green);
   --treeFocusedAccentColor: var(--colorTertiaryLighten3, green); */
 
-  --treeHoverBackgroundColor: var(--colorPrimaryDarken2);
-  --treeHoverAccentColor: var(--colorPrimaryLighten1);
+  --treeHoverBackgroundColor: var(--globalAppHeader);
```
Comment (amalyah) [inline src/css/common/_theme--dark.css:281]: See general comments for hover edits

File: src/css/common/_theme--dark.css [hunk 2]
```diff
@@ -269,23 +269,23 @@
...
+  --treeHoverAccentColor: var(--primary);
 
-  --treeSelectedBackgroundColor: var(--colorPrimaryDarken1);
+  --treeSelectedBackgroundColor: var(--surfaceElements);
```
Comment (amalyah) [inline src/css/common/_theme--dark.css:284]: See general comments for selected edits

File: src/css/common/_theme--light.css [hunk 3]
```diff
@@ -274,23 +274,20 @@
...
-  --treeHoverAccentColor: var(--colorSecondaryLighten3);
-
-  --treeSelectedBackgroundColor: var(
-    --colorSecondaryLighten3,
-    rgb(184, 222, 255)
-  );
-  --treeSelectedAccentColor: var(--colorSecondaryLighten1);
-  --treeSelectedTextColor: var(--colorBlack);
+  --treeTextColor: var(--defaultText);
+  --treeBackgroundColor: var(--inputBackground);
```
Comment (amalyah) [inline src/css/common/_theme--light.css:278]: surface elements color

File: src/css/common/_theme--light.css [hunk 4]
```diff
@@ -274,23 +274,20 @@
...
+  --treeBorderColor: var(--primaryDark);
+  --treeItemBorderColor: var(--primaryDark);
+  --treeAccentColor: var(--primary);
+
+  --treeHoverBackgroundColor: var(--backgroundColor);
```
Comment (amalyah) [inline src/css/common/_theme--light.css:283]: see general comments for hover states

File: src/css/common/_theme--light.css [hunk 5]
```diff
@@ -274,23 +274,20 @@
...
+  --treeHoverAccentColor: var(--primary);
+
+  --treeSelectedBackgroundColor: var(--inputBackground);
```
Comment (amalyah) [inline src/css/common/_theme--light.css:286]: surface element color
See general  comments for other selected state features

File: src/css/common/_theme--light.css [hunk 6]
```diff
@@ -274,23 +274,20 @@
...
+  --treeSelectedAccentColor: var(--primary);
+  --treeSelectedTextColor: var(--defaultText);
 
-  --treeChildrenBackgroundColor: var(--backgroundColor);
+  --treeChildrenBackgroundColor: var(--inputBackground);
```
Comment (amalyah) [inline src/css/common/_theme--light.css:290]: surface element color

File: src/css/components/tree.css [hunk 7]
```diff
@@ -107,13 +107,13 @@
 }
 
 .rux-tree .selected {
-  background-color: var(--treeSelectedBackgroundColor, rgb(0, 68, 107));
-  box-shadow: inset 5px 0 var(--treeSelectedAccentColor, rgb(77, 172, 255));
+  background-color: var(--treeSelectedBackgroundColor);
+  box-shadow: inset 5px 0 var(--treeSelectedAccentColor);
```
Comment (amalyah) [inline src/css/components/tree.css:111]: More shadows for your judgment @carmenparks (keep or not? design has none, code has some)

**Review by amalyah — CHANGES_REQUESTED** [review]

File: src/css/components/tree.css (hunk 7 above)
Comment (carmenparks) [reply src/css/components/tree.css:111]: Let's remove all shadows.

**Review by cskerbo — APPROVED** [review]

File: src/components/rux-tree/rux-tree.js [hunk 8]
```diff
@@ -334,7 +334,7 @@ export class RuxTree extends LitElement {
         font-size: 1rem;
 
         color: var(--treeTextColor, rgb(255, 255, 255));
```
[the remark this answers is not in the document]
Comment (im-cr) [reply src/components/rux-tree/rux-tree.js:336]: removed

File: src/css/common/_theme--light.css (hunk 6 above)
Comment (im-cr) [reply src/css/common/_theme--light.css:294]: updated

File: src/css/components/tree.css (hunk 7 above)
Comment (im-cr) [reply src/css/components/tree.css:118]: @amalyah @carmenparks this box shadow is used for creating the left border element when selected so it has to stay.

File: src/css/common/_theme--light.css (hunk 3 above)
Comment (im-cr) [reply src/css/common/_theme--light.css]: updated

File: src/css/common/_theme--dark.css (hunk 2 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css]: updated

File: src/css/common/_theme--dark.css (hunk 1 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css]: updated

**Comment by im-cr** [comment]
@amalyah hover state wasn't represented in designs, has been added in from your notes above.

**Comment by amalyah** [comment]
> @amalyah hover state wasn't represented in designs, has been added in from your notes above.

Sorry, it's built as an option in the filters, but it's not explicitly a symbol in the design files itself. I've attached an image to the Jira ticket to help for the desired  design state.

**Review by amalyah — CHANGES_REQUESTED** [review]
- Border design: Only selected rows should have the top and bottom primary-dark borders. For the primary items, they should have a bottom border of Background Color. If you want to see this clearer in the designs, I've attached a screenshot to the Jira ticket.

- Bug?: Not seeing a hover state over selected rows. (Which is okay if that's intentional)

- Hover: On hover, group items should change the caret to the same primary element font color as the text for now.

- Bug: When I open a group and  then hover another item, sometimes I see a bug where I see the correct hover state in the row I'm hovering over, but a hover background with incorrect  text colors on the previously opened folder. See the Jira ticket for screenshot.

**Comment by dmcalester** [comment]
@im-cr @amalyah @carmenparks I don't think the hover color is working. Aesthetically it’s really heavy and technically for whatever reason there’s a stair step effect on hover that was difficult to see in the previous version. Recommend for the time being removing hover altogether as no design spec available and revisiting in a dot-one

**Comment by amalyah** [comment]
> @im-cr @amalyah @carmenparks I don't think the hover color is working. Aesthetically it’s really heavy and technically for whatever reason there’s a stair step effect on hover that was difficult to see in the previous version. Recommend for the time being removing hover altogether as no design spec available and revisiting in a dot-one

Hi Duncan, I agree that hover states could stand another look and I agree that we should remove the stair step effect animation since that's feeling pretty awkward (meant to put in a comment, but forgot). However, it is in the design spec currently for both dark and light libraries. It is just currently a setting instead of a separate symbol. An example of that hover state in action was posted in the Jira ticket. Not sure if that affects your opinion on removing altogether from code or not, but I thought I'd state the correction.

**Comment by amalyah** [comment]
New bug found: double selection occurs sometimes between group items and individual items. Screenshot attached to Jira ticket for posterity.

**Comment by dmcalester** [comment]
> > @im-cr @amalyah @carmenparks I don't think the hover color is working. Aesthetically it’s really heavy and technically for whatever reason there’s a stair step effect on hover that was difficult to see in the previous version. Recommend for the time being removing hover altogether as no design spec available and revisiting in a dot-one
> 
> Hi @dmcalester , I agree that hover state designs should get another look and I agree that we should remove the stair step effect animation since that's feeling pretty awkward (meant to put in a comment, but forgot). However, the hover state is in the design spec currently for both dark and light libraries. It is just currently a setting instead of a separate symbol. An example of that hover state in action was posted in the Jira ticket. Not sure if that affects your opinion on removing altogether from code or not, but I thought I'd state the correction.

@amalyah @im-cr 
Understood. Let’s talk about this in standup because as-is it’s not ready for release.

**Comment by im-cr** [comment]
@amalyah double select issue resolved. Primary expanded item border color updated.

**Comment by amalyah** [comment]
> @amalyah double select issue resolved. Primary expanded item border color updated.

Looks like it's partially resolved. Still seeing the blue borders around some previously opened/closed folders. Screenshot on the Jira ticket.

Primary item border color should apply regardless of expanded/collapsed state and the blue dividing lines should be removed. Let me know if this would be easier to describe over a call.

**Review by amalyah — CHANGES_REQUESTED** [review]
See general comment

**Comment by im-cr** [comment]
@amalyah unable to reproduce last bug, let's hop on a call and review to close out.

**Review by dmcalester — APPROVED** [review]

**Review by amalyah — APPROVED** [review]

**Pull request merged**